com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -301,14 +303,27 @@ internal void ChangeOwnership(NetworkObject networkObject, ulong clientId)
                 throw new SpawnStateException("Object is not spawned");
             }
 
+            // Determine if the server is gong to lose ownership
+            var serverWillLoseOwnership = networkObject.OwnerClientId == NetworkManager.LocalClientId;
+
+            // Assign the new owner
             networkObject.OwnerClientId = clientId;
 
+            // Notify locally that the server lost ownership
+            if (serverWillLoseOwnership)
+            {
+                networkObject.InvokeBehaviourOnLostOwnership();
+            }
+
             networkObject.MarkVariablesDirty(true);
             NetworkManager.BehaviourUpdater.AddForUpdate(networkObject);
 
             // Server adds entries for all client ownership
             UpdateOwnershipTable(networkObject, networkObject.OwnerClientId);
 
+            // Always notify locally on the server when a new owner is assigned
+            networkObject.InvokeBehaviourOnGainedOwnership();
```

Comment:
This seems like a confusing choice to me. The other changes all seem to be geared toward notifying the server when ownership switches to/from server-owned... but this one is always notifying the server.

If the user is using this callback to try and track changes and know when the object becomes server owned, it could be really confusing that removing ownership notifies the server that it lost ownership, and then notifies the server immediately that it gained it back.

Reply:
Just pushed an update that assures it behaves like it always has where the server is notified of both lost ownership and gained ownership when ownership is changed.